Dockerfile

```diff
@@ -16,76 +15,99 @@ ENV CONDA_DIR /opt/conda
+                'ipywidgets=5.2*' \
+                'matplotlib=1.5*' \
+                'scipy=0.17*' \
+                'seaborn=0.7*' \
+                'cloudpickle=0.1*' \
+                statsmodels \
+                pandas \
+                'dill=0.2*' \
+                numpy \
+                jupyter \
+                notebook \
+                scikit-learn \
+    && $CONDA_DIR/bin/conda remove --quiet --yes --force qt pyqt \
+    && $CONDA_DIR/bin/conda clean -tipsy \
+    && fix-permissions.sh $CONDA_DIR \
+    && fix-permissions.sh $HOME 
+
+
+USER root
+
+# IPython
+EXPOSE 8888
+WORKDIR $HOME
+
+#&& chown $NB_UID:root /notebooks
+RUN mkdir /notebooks  \
+    && mkdir -p $HOME/.jupyter \
+    && echo "c.NotebookApp.ip = '*'" >> $HOME/.jupyter/jupyter_notebook_config.py \
+    && echo "c.NotebookApp.open_browser = False" >> $HOME/.jupyter/jupyter_notebook_config.py \
+    && echo "c.NotebookApp.notebook_dir = '/notebooks'" >> $HOME/.jupyter/jupyter_notebook_config.py \
     && yum erase -y gcc gcc-c++ glibc-devel \
     && yum clean all -y \
     && rm -rf /root/.npm \
     && rm -rf /root/.cache \
     && rm -rf /root/.config \
     && rm -rf /root/.local \
     && rm -rf /root/tmp \
-    && useradd -m -s /bin/bash -N -u $NB_UID $NB_USER \
-    && usermod -g root $NB_USER \
-    && chown -R $NB_USER $CONDA_DIR \
-    && conda remove --quiet --yes --force qt pyqt \
-    && conda remove --quiet --yes --force --feature mkl ; conda clean -tipsy
-
-
-ENV PATH /opt/conda/bin:$PATH
-
-ENV SPARK_HOME /opt/spark
-
-# Add a notebook profile.
-
-RUN mkdir /notebooks && chown $NB_UID:root /notebooks && chmod 1777 /notebooks
-
-EXPOSE 8888
-
-RUN mkdir -p -m 700 /home/$NB_USER/.jupyter/ && \
-    echo "c.NotebookApp.ip = '*'" >> /home/$NB_USER/.jupyter/jupyter_notebook_config.py && \
-    echo "c.NotebookApp.open_browser = False" >> /home/$NB_USER/.jupyter/jupyter_notebook_config.py && \
-    echo "c.NotebookApp.notebook_dir = '/notebooks'" >> /home/$NB_USER/.jupyter/jupyter_notebook_config.py && \
-    chown -R $NB_UID:root /home/$NB_USER && \
-    chmod g+rwX,o+rX -R /home/$NB_USER
-
-LABEL io.k8s.description="PySpark Jupyter Notebook." \
-      io.k8s.display-name="PySpark Jupyter Notebook." \
-      io.openshift.expose-services="8888:http"
-
-ENV TINI_VERSION v0.9.0
-ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
-ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini.asc /tini.asc
-RUN gpg --keyserver ha.pool.sks-keyservers.net --recv-keys 0527A9B7 && gpg --verify /tini.asc
-ADD start.sh /start.sh
+    && fix-permissions.sh /opt \
+    && fix-permissions.sh $CONDA_DIR \
+    && fix-permissions.sh /notebooks \
+    && fix-permissions.sh $HOME
 
-RUN chmod +x /tini /start.sh
 
-ENV HOME /home/$NB_USER
-USER $NB_UID
+ADD start.sh /usr/local/bin/start.sh
 WORKDIR /notebooks
+ENTRYPOINT ["tini", "--"]
+CMD ["fix-uid.sh", "/entrypoint", "start.sh"]
```

Comment:
fix-uid.sh does almost the same thing as /entrypoint, is it necessary?

https://github.com/radanalyticsio/openshift-spark/blob/master/scripts/spark/added/entrypoint